### PR TITLE
[DownstreamMerge] 4.12 initial merge from upstream: 7-18-22

### PR DESCRIPTION
**Pull request by trozet**
Conflicts in:
        both modified:   ../go-controller/cmd/ovnkube-trace/ovnkube-trace.go
        both modified:   ../go-controller/cmd/ovnkube/ovnkube.go
        both modified:   ../go-controller/go.mod
        both modified:   ../go-controller/go.sum
        both modified:   ../go-controller/hybrid-overlay/pkg/controller/master.go
        both modified:   ../go-controller/hybrid-overlay/pkg/controller/node_linux.go
        both modified:   ../go-controller/pkg/ovn/egressip_test.go
        both modified:   ../go-controller/pkg/ovn/master.go
        both modified:   ../go-controller/pkg/ovn/obj_retry.go
        both modified:   ../go-controller/pkg/ovn/pods_test.go
        both modified:   ../go-controller/pkg/ovn/policy.go
        both modified:   ../go-controller/pkg/ovn/policy_test.go
        both modified:   ../go-controller/pkg/util/ovs_unit_test.go
        both modified:   ../go-controller/vendor/modules.txt

All conflicts were resolved by moving to the upstream versions, except for:
        both modified:   ../go-controller/hybrid-overlay/pkg/controller/node_linux.go

### CODE REVIEW

**Comment by trozet** [comment]
ACL logging test is failing:
: [sig-network][Feature:Network Policy Audit logging] when using openshift ovn-kubernetes should ensure acl logs are created and correct [Suite:openshift/conformance/parallel] expand_more

File: go-controller/pkg/ovn/policy.go [hunk 1]
```diff
@@ -274,11 +338,11 @@ func buildDenyACLs(namespace, policy, pg, aclLogging string, policyType knet.Pol
 	denyMatch := getACLMatch(pg, "", policyType)
 	allowMatch := getACLMatch(pg, "(arp || nd)", policyType)
 	if policyType == knet.PolicyTypeIngress {
-		denyACL = buildACL(namespace, pg, policy, nbdb.ACLDirectionToLport, types.DefaultDenyPriority, denyMatch, nbdb.ACLActionDrop, aclLogging, policyType)
-		allowACL = buildACL(namespace, pg, "ARPallowPolicy", nbdb.ACLDirectionToLport, types.DefaultAllowPriority, allowMatch, nbdb.ACLActionAllow, "", policyType)
+		denyACL = buildACL(namespace, pg, ingressDefaultDenySuffix, nbdb.ACLDirectionToLport, types.DefaultDenyPriority, denyMatch, nbdb.ACLActionDrop, aclLogging, policyType)
```
Comment (jcaamano) [inline go-controller/pkg/ovn/policy.go:341]: This changed the ACL name and might be the reason why the ACL audit logging test is failing?

Comment (trozet) [reply go-controller/pkg/ovn/policy.go:341]: I think you are right. We will need to fix the test case before we can merge this PR.

Comment (trozet) [reply go-controller/pkg/ovn/policy.go:341]: Fixed by: https://github.com/openshift/origin/pull/27312

**Comment by dcbw** [comment]
https://github.com/openshift/origin/pull/27312 merged
/retest

**Comment by tssurya** [comment]
/retest
Still failing with audit logging, perhaps last tun did not pick up latest test binary

**Comment by tssurya** [comment]
/retest
again upgrade jobs are complaining about acl logging, but gcp/aws have passed :dancing_women:

**Comment by tssurya** [comment]
@trozet : upgrade still seems to be failing that test: https://prow.ci.openshift.org/view/gs/origin-ci-test/pr-logs/pull/openshift_ovn-kubernetes/1205/pull-ci-openshift-ovn-kubernetes-master-4.11-upgrade-from-stable-4.10-e2e-aws-ovn-upgrade/1550035600302673920 :/ wondering if I screwed up the update of the names from syncNetworkPolicy?... let me trigger an upgrade and see ...

**Comment by dcbw** [comment]
@trozet @tssurya if we update the origin testcase, could we print out the ACL line that fails? I forgot to ask for that first time around.

Looks like we're just failing on the two upgrade jobs right now... is that some backwards compat thing? IIRC Tim tried to account for that in the origin PR though.

**Comment by trozet** [comment]
ci/prow/4.11-upgrade-from-stable-4.10-e2e-aws-ovn-upgrade is failing the Audit log tests because it is using the 4.11 version of openshift-tests

@tssurya is going to work on updating the job

**Comment by tssurya** [comment]
https://github.com/openshift/release/pull/30710 should update the jobs (for new PRs)

**Comment by tssurya** [comment]
/lgtm
since the acl logging is the only job failing and we know why.

**Pull request merged**